presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/FloatingPointColumnChecksum.java

```diff
@@ -86,4 +101,63 @@ public String toString()
         String mean = (rowCount > 0 && sum != null) ? ", mean: " + ((double) sum / rowCount) : "";
         return format("sum: %s, NaN: %s, +infinity: %s, -infinity: %s%s", sum, nanCount, positiveInfinityCount, negativeInfinityCount, mean);
     }
+
+    // Class being used to return a result from the validate() method.
+    public static class ValidateResult
+    {
+        public boolean matched;
+        public final Optional<String> message;
+
+        public ValidateResult(boolean matched)
+        {
+            this.matched = matched;
+            this.message = Optional.empty();
+        }
+
+        public ValidateResult(boolean matched, Optional<String> message)
+        {
+            this.matched = matched;
+            this.message = message;
+        }
+    }
+
+    public static ValidateResult validate(
```

Comment:
Even changed to static function, I think it makes more sense to still put this function in [FloatingPointColumnValidator.java](https://github.com/prestodb/presto/pull/22143/files#diff-69d71b377216251d0931b7431965855d0a64922052e1167ad998c848f0b2cf5e)

FloatingPointColumnChecksum is only the data structure, FloatingPointColumnValidator will handle the validation logic

Reply:
@kewang1024 

Initially I kept it in the FloatingPointColumnValidator, but then I thought that this logic is better to be incapsulated in the structure that keeps the data. It already implements equals(), toString().
Make sense to put validate() here too, imho.

Let me know if you really prefer it back to the validator class.

Reply:
I think either all validation in should be in the respective checksum classes or none of them(also, might fit better in the validate class if it were a non-static method like equals e.g. `floatingPointChecksum.validate(FloatingPointChecksum other)`). As things stand it seems a bit out of place, and I agree with @kewang1024 that it would be better in the validate class.